src/mlpack/methods/reinforcement_learning/environment/acrobat.hpp

```diff
@@ -0,0 +1,319 @@
+#define MLPACK_METHODS_RL_ENVIRONMENT_ACROBAT_HPP
+#define PIE 3.14159265358979323846
+
+#include <math.h>
+#include <random>
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack{
+namespace rl{
+/*
+ * Implementation of Acrobat game
+ * Acrobot is a 2-link pendulum with only the second joint actuated
+ * Intitially, both links point downwards. The goal is to swing the
+ * end-effector at a height at least the length of one link above the base.
+ * Both links can swing freely and can pass by each other, i.e., they don't
+ * collide when they have the same angle.
+ */
+class Acrobat
+{
+ public:
+   /*
+    * Implementation of Acrobat State
+    */    
+  class State
+  {
+   public :
+    /**
+    * Construct a state instance.
+    */
+    State(): data(dimension)
+    { /* nothing to do here */ }
+    /**
+    * Construct a state instance from given data.
+    *
+    * @param data Data for the acrobat.
+    * Consists of sin() and cos() for two rotational joint and angular velocity.
+    * 
+    */
+    State(const arma::colvec& data): data(data)
+    { /* nothing to do here */ }
+
+    //! Modify the state representation
+    arma::colvec& Data() {return data;}
+
+    //! Get value of theta1
+    double Theta1() const {return data[0];}
+    //! Modify value of theta1
+    double& Theta1() {return data[0];}
+
+    //! Get value of theta2
+    double Theta2() const {return data[1];}
+    //! Modify value of theta2
+    double& Theta2() {return data[1];}
+
+    //! Get value of Angular velocity 1
+    double AngularVelocity1() const { return data[2]; }
+    //! Modify the angular velocity 1.
+    double& AngularVelocity1() { return data[2]; }
+
+    //! Get value of Angular velocity 2
+    double AngularVelocity2() const { return data[3]; }
+    //! Modify the angular velocity 2.
+    double& AngularVelocity2() { return data[3]; }
+
+    //! Encode the state to a column vector.
+    const arma::colvec& Encode() const { return data; }
+
+    //! Dimension of the encoded state.
+    static constexpr size_t dimension = 4;
+
+   private :
+    arma::colvec data;
+  };
+  /* 
+   * Implementation of action for cartpole
+   */
+  enum Action
+  {
+    negativeTorque,
+    zeroTorque,
+    positiveTorque,
+    // Track the size of the action space.
+    size
+  };
+
+   /**
+   * Construct a Cart Pole instance using the given constants.
+   *
+   * @param gravity gravity
+   * @param link_length1 length of link 1.
+   * @param link_length2 length of link 2.
+   * @param link_mass1 mass of link 1.
+   * @param link_mass2 mass of link 2.
+   * @param link_com1 position of the center of mass of link 1.
+   * @param link_com2 position of the center of mass of link 2.
+   * @param link_moi moments of inertia for both link.
+   * @param max_vel1 max angular velocity of link1.
+   * @param max_vel2 max angular velocity of link2.
+   */
+  Acrobat(const double gravity = 9.81,
+          const double link_length1 = 1.0,
```

Comment:
Use camel casing for all names; see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#naming-conventions.